crates/phactory/src/prpc_service.rs

```diff
@@ -815,4 +822,174 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> PhactoryApi
         let echo_msg = request.echo_msg;
         Ok(pb::EchoMessage { echo_msg })
     }
+
+    fn get_worker_key_challenge(&mut self, _request: ()) -> RpcResult<pb::WorkerKeyChallenge> {
+        let mut phactory = self.lock_phactory();
+        let system = phactory.system()?;
+        let challenge = system.get_worker_key_challenge();
+        Ok(pb::WorkerKeyChallenge::new(challenge))
+    }
+
+    fn get_worker_key(
+        &mut self,
+        request: pb::WorkerKeyChallengeResponse,
+    ) -> RpcResult<pb::GetWorkerKeyResponse> {
+        let mut phactory = self.lock_phactory();
+        let system = phactory.system()?;
+        let my_identity_key = system.identity_key.clone();
+
+        // 1. verify RA report
```

Comment:
We need the timestamp in the RA report from Intel as the trusted time source, so we can know the key exporter pRuntime must be up-to-date. Otherwise, attackers can deliberately keep some old version vulnerable pRuntimes, and dump its WorkerKey with tx replaying.

Reply:
If we really wants to enforce local attestation, we can create a seal-to-mrsigner file with a nonce, and check the nonce in another pruntime.

Reply:
This is not "local" to me since they can move the file anywhere